pennylane/devices/preprocess.py

```diff
@@ -756,7 +756,7 @@ def _get_diagonalized_tape_and_wires(tape):
     """Apply the diagonalize_measurements transform to the tape and extract a list of
     all the wires present in the measurements"""
 
-    (diagonalized_tape,), _ = qml.transforms.diagonalize_measurements(tape)
+    (diagonalized_tape,), _ = qml.transforms.diagonalize_measurements(tape, use_op_gates_only=True)
```

Comment:
This is something we added just on this branch to allow us to force the `diagonalize_measurement` transform to use the diagonalizing gates on the operations instead of using RY and RX (non-Clifford gates that artificially limit how much of the circuit can be handled with offline corrections)